### ISSUE TITLE
Macbook can't get network card information

### ISSUE THREAD

**Issue by muddlelife** [rikonaka/pistol-rs#12]
Hello, I am using Macbook M1, I found that when using pistol to send syn packets, the source IP address could not be automatically obtained. I also saw in your code that you did not obtain the network card for macOS, could you add this function? If you need my help, please tell me at any time

https://github.com/rikonaka/pistol-rs/blob/013b27543fd86fdb8306385b4a6ff61f200aa29c/src/layers.rs#L610

**Comment by rikonaka** [comment]
In fact, the biggest problem right now is that I don't have any Apple products to use for testing 😂.

**Comment by muddlelife** [comment]
I found crates [local-ip-address](https://crates.io/crates/local-ip-address) to be very useful, and in my tests it worked perfectly to get my current network card, both linux and Windows supported

``` rust
use local_ip_address::local_ip;

fn main() {
    let my_local_ip = local_ip().unwrap();

    println!("This is my local IP address: {:?}", my_local_ip);
}
```

```text
This is my local IP address: 192.168.47.242
```
And I saw his source code and I thought this would be a good fit

**Comment by rikonaka** [comment]
I took a look at the library, but realized that it can only be used to return local ip addresses and not route addresses.

**Comment by rikonaka** [comment]
Hi @muddlelife , I added the adaptation for Unix-like systems because I don't have any Apple products, but I tested it on FreeBSD and the results showed no problems, could you please test it on macOS?

![FreeBSD test](https://github.com/rikonaka/pistol-rs/assets/13602602/061bf03e-2043-4ec4-b190-6217fda43bb8)

**Comment by muddlelife** [comment]
Here are the results of my tests

<img width="1100" alt="image" src="https://github.com/rikonaka/pistol-rs/assets/79511104/c891b6bc-4533-4518-b99a-4f44861bfdbd">

But I used the latest pistol-rs crates, version 1.0.6, and still couldn't scan on MacOS, with the following error
```text
warning: `portscan` (bin "portscan") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/portscan`
thread 'main' panicked at src/main.rs:27:7:
called `Result::unwrap()` on an `Err` value: Invalid component in a MAC address string
```
Here's my test code

Cargo.toml
```toml
[package]
name = "portscan"
version = "0.1.0"
edition = "2021"

[dependencies]
anyhow = "1.0.86"
pistol = "1.0.6"

[dependencies.pnet]
version = "0.35.0"
```
main.rs
```rust
use pistol::{tcp_syn_scan, Host, Target};
use std::net::Ipv4Addr;
use std::time::Duration;
use anyhow::Result;

fn main() -> Result<()> {
    // When using scanning, please use a real local address to get the return packet.
    // And for flood attacks, please consider using a fake address.
    // If the value here is None, the programme will automatically look up the available addresses from the existing interfaces on the device.
    let src_ipv4 = Some(Ipv4Addr::new(192, 168, 47, 242));
    // If the value of `source port` is `None`, the program will generate the source port randomly.
    let src_port = None;
    // The destination address is required.
    let dst_ipv4 = Ipv4Addr::new(192, 168, 100, 17);
    let threads_num = 8;
    let timeout = Some(Duration::new(3, 0));
    // Test with an open port `22` and a closed port `99`.
    let host = Host::new(dst_ipv4, Some(vec![22, 99]));
    /// Users should build the `target` themselves.
    let target = Target::new(vec![host]);
    let ret = tcp_syn_scan(
        target,
        src_ipv4,
        src_port,
        threads_num,
        timeout,
    ).unwrap();
    println!("{}", ret);
    Ok(())
}
```
And I can ping 192.168.100.17
<img width="972" alt="image" src="https://github.com/rikonaka/pistol-rs/assets/79511104/68680c25-4708-48f8-b1c5-d5efe31fb48d">

**Comment by rikonaka** [comment]
Ok, I had fixed this problem and released the v1.0.7, please test it again. 😉
